src/gluonts/distribution/distribution.py

```diff
@@ -253,6 +253,30 @@ def quantile(self, level: Tensor) -> Tensor:
         """
         raise NotImplementedError()
 
+    def slice(self, begin: Tuple, end: Tuple) -> "Distribution":
```

Comment:
Actually, this method may not be useful in general, and could be removed. The reason is that not all tensors used to construct a distribution need to have the same number of axes: this is the case for `MultivariateGaussian` for example. Therefore, the `begin` and `end` tuples may not be usable across all constructor arguments.

Removing this and tests.